src/get-installation-authentication.ts

```diff
@@ -16,12 +16,18 @@ export async function getInstallationAuthentication(
   const installationId = (options.installationId ||
     state.installationId) as number;
 
-  if (typeof installationId !== "number") {
+  if (typeof installationId === "undefined") {
     throw new Error(
       "[@octokit/auth-app] installationId option is required for installation authentication."
     );
   }
 
+  if (typeof installationId !== "number") {
+    throw new Error(
+      "[@octokit/auth-app] installationId option should be of type number."
+    );
+  }
```

Comment:
I don't think we need to throw this error. An `installationId` string value should work just fine, if it doesn't, that is something we should fix.

What we could do instead is to check if `installationId` is truthy above. It can never by `0`, so there shouldn't be a problem with that. So instead of

```js
if (typeof installationId !== "number") {
```

We could do

```js
if (!installationId) {
```